### PR TITLE
[KEYCLOAK-12107] Add tests for Applications page

### PR DESCRIPTION
**Pull request by douglaspalmer**

### CODE REVIEW

File: testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/page/ApplicationsPage.java
Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/page/ApplicationsPage.java]: I'm not sure we should use selectors like that. This will be very error prone in the future when slightly changing the layout or even updating PF/React. Please add IDs where missing. (To keep all IDs unique, you can make a part of the ID dynamic like e.g. [here](https://github.com/keycloak/keycloak/blob/210fd92d2328b3d8173d7c07e2708e6d9544a396/themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/device-activity-page/DeviceActivityPage.tsx#L283)).

File: testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/page/ApplicationsPage.java
Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/page/ApplicationsPage.java]: I'm not sure we should use selectors like that.

File: testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/ApplicationsTest.java
Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/ApplicationsTest.java]: We should also test collapsing/expanding of an app works.

File: testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/page/ApplicationsPage.java
Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/page/ApplicationsPage.java]: Please use [`getTextFromElement`](https://github.com/keycloak/keycloak/blob/f426643225337f17f640d43e0f5b236f09d70e9d/testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/util/UIUtils.java#L159) to avoid any browser-related problems in the future.

**Review by vmuzikar — COMMENTED** [review]
Thanks @douglaspalmer!

I did a brief review and overall looks good – for the tech preview (and given the time we have) it would be sufficient. However, I found a few issues. I'll leave it up to you if you fix it now or as a follow-up. ;)

File: testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/page/ApplicationsPage.java
Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/page/ApplicationsPage.java]: I'm not sure how this can work. The ID doesn't have a unique value at all so we cannot be sure what app was selected. This is of course not an issue with the test as such but rather with the page.

File: testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/ApplicationsTest.java
Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/ApplicationsTest.java]: We might want to try toggling only one app. What if the button is broken and toggles all apps at once instead of only one? :)

**Review by vmuzikar — CHANGES_REQUESTED** [review]
@douglaspalmer Thanks for the update! Just a few minor-ish comments, other than that looks good.

File: themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/applications-page/ApplicationsPage.tsx
Comment (vmuzikar) [inline themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/applications-page/ApplicationsPage.tsx]: One last thing. This (`aria-controls`) should have the same value as this:
https://github.com/keycloak/keycloak/blob/ab862c045231dffaba342b400cf301d409aedc47/themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/applications-page/ApplicationsPage.tsx#L152
I think this has something to do with accessibility. Even though we'll address accessibility in general later, let's not add more issues. ;)

**Comment by abstractj** [comment]
Just waiting for Travis, but it LGTM

**Review by abstractj — APPROVED** [review]

**Pull request merged**
